src/index.ts

```diff
@@ -120,16 +120,35 @@ const createProxyHandler = <T extends object>(origObj: T, frozen: boolean) => {
 };
 
 /**
- * create a proxy
+ * Create a proxy.
  *
- * It will recursively create a proxy upon access.
+ * This function will create a proxy at top level and proxy nested objects as you access them,
+ * in order to keep track of which properties were accessed via get/has proxy handlers:
+ *
+ * NOTE: Printing of WeakMap is hard to inspect and not very readable
+ * for this purpose you can use the `affectedToPathList` helper.
+ *
+ * @param {object} obj - Object that will be wrapped on the proxy.
+ * @param {WeakMap<object, unknown>} affected -
+ * WeakMap that will hold the tracking of which properties in the proxied object were accessed.
+ * @param {WeakMap<object, unknown>} [proxyCache] -
+ * WeakMap that will help keep referential identity for proxies.
+ * @returns {Proxy<object>} - Object wrapped in a proxy.
  *
  * @example
  * import { createDeepProxy } from 'proxy-compare';
  *
- * const obj = ...;
- * const affected = new WeakMap();
- * const proxy = createDeepProxy(obj, affected);
+ * const orginal = { a: "1", c: "2", d: { e: "3" } };
+ * const affecteProps = new WeakMap();
```

Comment:
do you mean `affectedProps`? or it is the word i never used..

btw, it's not technically props it's `<obj, props>`

Reply:
yeah `affecteProps` it was a typo. ill update the const name to `affected` to avoid any confusion 💯

Reply:
done